terraform-plans/files/github/charm_promote.yaml

```diff
@@ -1,14 +1,22 @@
-name: Promote charm to other tracks and channels
+name: Promote charm to default track, standard risk levels.
 
 on:
   workflow_dispatch:
     inputs:
       destination-channel:
         description: 'Destination Channel, e.g. latest/candidate'
         required: true
+        type: choice
+        options:
+          - 'latest/candidate'
+          - 'latest/stable'
       origin-channel:
         description: 'Origin Channel, e.g. latest/edge'
         required: true
+        type: choice
+        options:
+          - 'latest/edge'
+          - 'latest/candidate'
```

Comment:
What do you think about using something like this:
```yaml
options:
          - 'latest/edge -> latest/candidate'
          - 'latest/candidate -> latest/stable'
```
I know that implementation would require some changes, but it will be more bullet proof for user.

Reply:
The implementation will be something like:


```
name: Promote charm to default track, standard risk levels.

on:
  workflow_dispatch:
    inputs:
      channel-promotion:
        description: 'Channel Promotion, e.g. latest/edge -> latest/candidate'
        required: true
        type: choice
        options:
          - 'latest/edge -> latest/candidate'
          - 'latest/candidate -> latest/stable'

jobs:
  promote-charm:
    name: Promote charm
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Set channels
        id: set-channels
        run: |
          if [[ "${{ github.event.inputs.channel-promotion }}" == 'latest/edge -> latest/candidate' ]]; then
            echo "destination-channel=latest/candidate" >> $GITHUB_ENV
            echo "origin-channel=latest/edge" >> $GITHUB_ENV
          elif [[ "${{ github.event.inputs.channel-promotion }}" == 'latest/candidate -> latest/stable' ]]; then
            echo "destination-channel=latest/stable" >> $GITHUB_ENV
            echo "origin-channel=latest/candidate" >> $GITHUB_ENV
          fi
      - name: Release charm to channel
        uses: canonical/charming-actions/release-charm@2.4.0
        with:
          credentials: ${{ secrets.CHARMHUB_TOKEN }}
          github-token: ${{ secrets.GITHUB_TOKEN }}
          destination-channel: ${{ env.destination-channel }}
          origin-channel: ${{ env.origin-channel }}
```

I haven't test it, how do you think @Pjack ?

Reply:
This is my preferred user behavior. If the code is maintainable, then I'm happy with this.

Reply:
> `origin, destination = channel-promotion.split('->').map(strip_whitespace)`

I don't think this is doable on workflow yaml.

Reply:
I put the implementation on latest commit. Please check it there.